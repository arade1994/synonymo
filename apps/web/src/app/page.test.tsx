import { describe, expect, test } from "vitest";

import { render, screen } from "@testing-library/react";

import Home from "./page";

describe("HomePage", () => {
  test("It should render and match a snapshot by default", () => {
    const { baseElement } = render(<Home />);

    expect(baseElement).toMatchSnapshot();
  });

  test("It should render links for adding and searching synonyms", () => {
    render(<Home />);

    const addSynonymsLink = screen.getByTestId("addSynonymsLink");
    const searchSynonymsLink = screen.getByTestId("searchSynonymsLink");

    expect(addSynonymsLink).toBeDefined();
    expect(addSynonymsLink.innerHTML).toEqual("Add Synonyms");
    expect(searchSynonymsLink).toBeDefined();
    expect(searchSynonymsLink.innerHTML).toEqual("Search Synonyms");
  });
});
