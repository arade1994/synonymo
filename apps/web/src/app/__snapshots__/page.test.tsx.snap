// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage > It should render and match a snapshot by default 1`] = `
<body>
  <div>
    <main
      class="_main_6b97ca"
    >
      <div
        class="_content_6b97ca"
      >
        <h1
          class="_title_6b97ca"
        >
          Welcome to 
          <span
            class="_highlight_6b97ca"
          >
            Synonymo
          </span>
        </h1>
        <p
          class="_description_6b97ca"
        >
          The fun and easy way to build your vocabulary. Add new words and discover their synonyms, or search for existing ones to expand your lexicon.
        </p>
        <div
          class="_ctaContainer_6b97ca"
        >
          <a
            class="_ctaButton_6b97ca"
            href="/synonyms"
            id="addSynonymsLink"
          >
            Add Synonyms
          </a>
          <a
            class="_ctaButtonSecondary_6b97ca"
            href="/synonyms/search"
            id="searchSynonymsLink"
          >
            Search Synonyms
          </a>
        </div>
      </div>
    </main>
  </div>
</body>
`;
